code/datums/components/beauty.dm

```diff
@@ -11,13 +11,13 @@
 	enter_area(null, A)
 
 /datum/component/beauty/proc/enter_area(datum/source, area/A)
-	if(A.outdoors)
+	if(!istype(A) || A.outdoors)
```

Comment:
Don't istype here, check if A exists before your non signal calls on this proc. The signal version will always have the proper area provided